src/structures/tickets.rs

```diff
@@ -66,27 +70,70 @@ macro_rules! impl_ticket_try_froms {
 
                 let hierarchy = Hierarchy::try_from(TpmHandle::try_from(tss_ticket.hierarchy)?)?;
 
-                Ok($ticket_type { hierarchy, digest })
+                Ok($ticket_type {
+                    tag,
+                    hierarchy,
+                    digest,
+                })
             }
         }
     };
 }
 
 pub trait Ticket {
-    const TAG: StructureTag;
+    const POSSIBLE_TAGS: &'static [StructureTag];
+    fn tag(&self) -> StructureTag {
+        if Self::POSSIBLE_TAGS.len() == 1 {
+            Self::POSSIBLE_TAGS[0]
+        } else {
+            unimplemented!("Multiple possible tags, but no tag() override");
```

Comment:
Would wonderful if that could be a compile time error ...

Reply:
Right, but this is an implementation in our code, over where we just need to make sure that that's defined (and with this change, we have defined all 4 ticket types).

Reply:
I could just not have a default implementation, and always force an implementation of it.

Reply:
> Right, but this is an implementation in our code, over where we just need to make sure that that's defined (and with this change, we have defined all 4 ticket types).

True, since it's based on a spec that has a fixed number of cases, I think just visually checking should be fine

Reply:
So, are we okay with the current approach? If you still don't feel comfortable (Which I understand, I was also hesitant to do it this way), I'm happy to just implement the `fn tag(&self) -> StructureTag` for each of the 3 remaining implementations. They're all very simple, so the code reuse is not that big of a deal.

Reply:
I say implement it for each ticket. I think that is nicer.

Reply:
I think I'm ok either way, as long as it's a conscious decision and we check them thoroughly! Maybe having a small unit test covering each ticket type would be good

Reply:
I implemented it for each ticket.